### PR TITLE
samples(spanner): Add samples for Postgres support in Spanner

### PR DESCRIPTION
**Pull request by SandeepTuniki**
Code samples for Spangres support

## Checklist
- [ ] **Tests** pass
- [ ] **Lint** pass: `bundle exec rubocop`
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: spanner/spanner_postgresql_order_nulls.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    ordered_names << row[:FirstName]
+  end
+
+  # This returns the singers in order Bruce, Alice, null,
+  results = client.execute "SELECT FirstName FROM Singers ORDER BY FirstName NULLS LAST"
+  results.rows.each do |row|
+    ordered_names << row[:FirstName]
+  end
+
+  ordered_names
```
Comment (NivedhaSenthil) [inline spanner/spanner_postgresql_order_nulls.rb:71]: We generally do not return values from sample, can we show whats in the ordered_names instead here ? 
That will give an idea of what we trying show

File: spanner/spec/spanner_postgresql_cast_data_type_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    capture do
+      spanner_postgresql_cast_data_type project_id: @project_id,
+                                        instance_id: @instance_id,
+                                        database_id: @database_id
+    end
+
+    expect(captured_output).to include "1"
+    expect(captured_output).to include "2"
+    expect(captured_output).to include "3"
+    expect(captured_output).to include "4" # How to test this? It is returned as bytes
```
Comment (NivedhaSenthil) [inline spanner/spec/spanner_postgresql_cast_data_type_spec.rb:40]: Please remove this comment

**Review by NivedhaSenthil — CHANGES_REQUESTED** [review]

File: spanner/spanner_connect_postgresql_database.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def spanner_connect_postgresql_database project_id:, instance_id:, database_id:
+  # [START spanner_connect_postgresql_database]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
```
[the remark this answers is not in the document]
Comment (SandeepTuniki) [reply spanner/spanner_connect_postgresql_database.rb]: Changed for all the samples.

File: spanner/spanner_postgresql_order_nulls.rb (hunk 1 above)
Comment (SandeepTuniki) [reply spanner/spanner_postgresql_order_nulls.rb]: Oops, my mistake. I changed the return statement to print statements and ensured the tests captured the output.

File: spanner/spec/spanner_postgresql_cast_data_type_spec.rb (hunk 2 above)
Comment (SandeepTuniki) [reply spanner/spec/spanner_postgresql_cast_data_type_spec.rb]: Done.

File: spanner/spanner_postgresql_batch_dml.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    row_counts = transaction.batch_update do |b|
+      b.batch_update "INSERT INTO Singers (SingerId, FirstName, LastName) VALUES ($1, $2, $3)",
+                     params: { p1: 3, p2: "Olivia", p3: "Garcia" }
+    end
+  end
+
+  statement_count = row_counts.count
+
+  puts "Executed #{statement_count} SQL statements using Batch DML."
+end
```
Comment (NivedhaSenthil) [inline spanner/spanner_postgresql_batch_dml.rb:37]: seems method call to make the sample executable is missing in few samples, please add them

``` ruby
if $PROGRAM_NAME == __FILE__
  postgresql_batch_dml project_id: ARGV.shift, instance_id: ARGV.shift, database_id: ARGV.shift
end
```

Comment (SandeepTuniki) [reply spanner/spanner_postgresql_batch_dml.rb:37]: Added them to add the files.

**Comment by NivedhaSenthil** [comment]
LGTM

**Review by NivedhaSenthil — APPROVED** [review]

**Comment by SandeepTuniki** [comment]
samples(spanner): Add samples for Postgres support in Spanner

**Pull request merged**